### PR TITLE
FIX Respect searchable_fields

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-framework/issues/10683

Installer run - https://github.com/emteknetnz/silverstripe-installer/actions/runs/4485413512

### CODE REVIEW

File: src/Forms/GridField/GridFieldFilterHeader.php [hunk 1]
```diff
@@ -254,21 +254,29 @@ public function getManipulatedData(GridField $gridField, SS_List $dataList)
...
-        $columns = $gridField->getColumns();
-        foreach ($columns as $columnField) {
-            $metadata = $gridField->getColumnMetadata($columnField);
-            $title = $metadata['title'];
-
-            if ($title && $list->canFilterBy($columnField)) {
-                return true;
+        $modelClass = $gridField->getModelClass();
+        $searchableFields = Config::inst()->get($modelClass, 'searchable_fields');
+        if (!empty($searchableFields)) {
```
Comment (michalkleiner) [inline src/Forms/GridField/GridFieldFilterHeader.php]: Do we want to make sure here that `searchable_fields` is an array if set?

File: src/Forms/GridField/GridFieldFilterHeader.php [hunk 2]
```diff
@@ -254,21 +254,29 @@ public function getManipulatedData(GridField $gridField, SS_List $dataList)
...
-
-        $columns = $gridField->getColumns();
-        foreach ($columns as $columnField) {
-            $metadata = $gridField->getColumnMetadata($columnField);
-            $title = $metadata['title'];
-
-            if ($title && $list->canFilterBy($columnField)) {
-                return true;
+        $modelClass = $gridField->getModelClass();
+        $searchableFields = Config::inst()->get($modelClass, 'searchable_fields');
```
Comment (GuySartorelli) [inline src/Forms/GridField/GridFieldFilterHeader.php]: Searchable fields should be fetched by calling `searchableFields()` on a singleton, i.e. `$modelClass::singleton()->searchableFields()`
See [SearchContext](https://github.com/silverstripe/silverstripe-framework/blob/a4929a171e0e0956ecff6f49b04a88f1ec9c26f9/src/ORM/Search/SearchContext.php#L164-L165) and [GridFieldAddExistingAutocompleter](https://github.com/silverstripe/silverstripe-framework/blob/a4929a171e0e0956ecff6f49b04a88f1ec9c26f9/src/Forms/GridField/GridFieldAddExistingAutocompleter.php#L340-L342) as examples of where this is already done.

File: src/Forms/GridField/GridFieldFilterHeader.php [hunk 3]
```diff
@@ -254,21 +254,29 @@ public function getManipulatedData(GridField $gridField, SS_List $dataList)
...
+        if (!empty($searchableFields)) {
+            // Scenario #1 - searchable_fields is defined on model class
+            foreach ($searchableFields as $searchableField) {
+                if ($list->canFilterBy($searchableField)) {
+                    return true;
+                }
+            }
```
Comment (GuySartorelli) [inline src/Forms/GridField/GridFieldFilterHeader.php]: `canFilterBy()` is fine for `summary_fields()` - but we can declare `searchable_fields` that aren't _traditionally_ filterable here, and then tell `SearchContext` how to search using those fields. If a field is in searchable fields that _isn't_ in summary fields, we should just assume it's filterable and let `SearchContext` sort it out.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: src/Forms/GridField/GridFieldFilterHeader.php (hunk 3 above)
Comment (emteknetnz) [reply src/Forms/GridField/GridFieldFilterHeader.php]: Rewrote the method

File: src/Forms/GridField/GridFieldFilterHeader.php (hunk 2 above)
Comment (emteknetnz) [reply src/Forms/GridField/GridFieldFilterHeader.php]: Updated

File: src/Forms/GridField/GridFieldFilterHeader.php (hunk 1 above)
Comment (emteknetnz) [reply src/Forms/GridField/GridFieldFilterHeader.php]: Have changed to use $modelClass::singleton()->searchableFields() which returns an array

**Review by GuySartorelli — APPROVED** [review]
Works as expected

**Pull request merged**